superset-frontend/src/views/components/MenuRight.tsx

```diff
@@ -113,9 +148,41 @@ const RightMenu = ({
             }
             icon={<Icons.TriangleDown />}
           >
-            {dropdownItems.map(
-              menu =>
-                findPermission(menu.perm, menu.view, roles) && (
+            {dropdownItems.map(menu => {
+              if (menu.label === 'Data') {
```

Comment:
I'd suggest keeping the rendering and the data separate. Can we instead maybe use menu.childs as a requirement instead of the specific menu.label ==="Data" and then iterate over the childs menus?